core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -94,9 +95,10 @@ public void filter(ContainerRequestContext request)
             request.abortWith(Response.seeOther(DISABLED_LOCATION_URI).build());
             return;
         }
-        Optional<Map<String, Object>> claims;
         Optional<TokenPair> tokenPair = getTokenPair(request);
-        claims = tokenPair.flatMap(this::getAccessTokenClaims);
+        Optional<Map<String, Object>> claims = tokenPair
+                .filter(this::tokenNotExpired)
```

Comment:
Will it ever happen? IIRC JWT library throws an exception when trying to parse an expired token.

Reply:
This checks is for the access token not for the token we mint. Additionally it reduces the load on the IdP in case of expired access token.